File: 06_Inheritance/MultipleInheritance.cpp
#include <iostream>
using namespace std;

class Animal {
public:
    virtual void eat() = 0; 
};

class Mammal {
public:
    void sleep() {
        cout << "This mammal sleeps." << endl;
    }
};

class Dog : public Mammal, public Animal {
public:
    void eat() override {
        cout << "The dog eats." << endl;
    }

    void bark() {
        cout << "The dog barks." << endl;
    }
};

int main() {
    Dog dog;
    dog.sleep();  
    dog.eat();    
    dog.bark();   
    return 0;
}

File: 06_Inheritance/MultipleInheritance.java
interface Animal {
    void eat();
}

class Mammal {
    void sleep() {
        System.out.println("This mammal sleeps.");
    }
}

class Dog extends Mammal implements Animal {
    public void eat() {
        System.out.println("The dog eats.");
    }

    void bark() {
        System.out.println("The dog barks.");
    }
}

public class MultipleInheritance {
    public static void main(String[] args) {
        Dog dog = new Dog();
        dog.sleep();  
        dog.eat();   
        dog.bark();   
    }
}
